src/read.c

```diff
@@ -1953,61 +1953,35 @@ static avifBool avifParseImageGridBox(avifImageGrid * grid,
 
 #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
 
-static avifBool avifParseToneMappedImageBox(avifGainMapMetadata * metadata,
-                                            const uint8_t * raw,
-                                            size_t rawLen,
-                                            avifBool * isUnsupportedVersion,
-                                            avifDiagnostics * diag)
+static avifBool avifParseGainMapMetadta(avifGainMapMetadata * metadata, avifROStream * s)
```

Comment:
Nit/optional: Add the `writerVersion` parameter to be future-proof. It will be ignored in the current implementation, so we will need to add a `(void)` cast.

We can also wait until this is really needed.

Reply:
The remaining data check could be moved to this function. Conceptually it seems to belong in this function.

If so, then this function will use the `writerVersion` parameter. But we will also need to pass `supportedMetadataVersion` to this function or make it a constant in the file scope, so I'm not sure if this is a good idea.

Reply:
Thanks for the suggestions, but I think we can keep things simple (and therefore more readable) for now.